### ISSUE TITLE
GET on /info or /declare endpoint cause BIG-IP to go into "Changes Pending" in HA

### ISSUE THREAD

**Issue by euphario** [F5Networks/f5-appsvcs-extension#391]
### Environment
 * Application Services Version: 3.24.0
 * BIG-IP Version: 15.1.2
 * BIG-IP HA pair

### Summary
GET calls to /info or /declare endpoints of AS3 can put BIG-IP into an out-of-sync state in a HA pair.

### Steps To Reproduce
```
1. GET /mgmt/shared/appsvcs/info
```
This may cause BIG-IP to got into "Changes Pending", typically on the first request, and after making a GET request to the /declare endpoint

```
2. GET /mgmt/shared/appsvcs/declare
```
This always put BIG-IP into an out-of-sync state with its peer.

Audit logs;
```
Dec  2 15:50:14 bigip2.localhost.localdomain info httpd(pam_audit)[15299]: 01070417:6: AUDIT - user admin - RAW: httpd(pam_audit): user=admin(admin) partition=[All] level=Administrator tty=(unknown) host=172.18.7.114 attempts=1 start="Wed Dec  2 15:50:14 2020" end="Wed Dec  2 15:50:14 2020".
Dec  2 15:50:15 bigip2.localhost.localdomain notice icrd_child[21479]: 01420002:5: AUDIT - pid=21479 user=admin folder=/Common module=(tmos)# status=[01020037:3: The requested folder (/Common/appsvcs) already exists.] cmd_data=create sys folder appsvcs
Dec  2 15:50:15 bigip2.localhost.localdomain notice icrd_child[21479]: 01420002:5: AUDIT - pid=21479 user=admin folder=/Common module=(tmos)# status=[01020037:3: The requested folder (/Common/appsvcs) already exists.] cmd_data=create sys folder appsvcs
Dec  2 15:50:15 bigip2.localhost.localdomain notice icrd_child[21479]: 01420002:5: AUDIT - pid=21479 user=admin folder=/Common module=(tmos)# status=[01020066:3: The requested value list (/Common/appsvcs/dataStore) already exists in partition Common.] cmd_data=create ltm data-group internal /Common/appsvcs/dataStore { type string }
Dec  2 15:50:15 bigip2.localhost.localdomain notice icrd_child[21479]: 01420002:5: AUDIT - pid=21479 user=admin folder=/Common module=(tmos)# status=[01020066:3: The requested value list (/Common/appsvcs/dataStore) already exists in partition Common.] cmd_data=create ltm data-group internal /Common/appsvcs/dataStore { type string }
Dec  2 15:50:15 bigip2.localhost.localdomain notice mcpd[7403]: 01070417:5: AUDIT - client tmsh, tmsh-pid-21479, user admin - transaction #224744-2 - object 0 - modify { value_list { value_list_name "/Common/appsvcs/dataStore" value_list_config_source 0 } } [Status=Command OK]
Dec  2 15:50:15 bigip2.localhost.localdomain notice mcpd[7403]: 01070417:5: AUDIT - client tmsh, tmsh-pid-21479, user admin - transaction #224744-3 - object 0 - obj_delete { class_string_item { class_string_item_name "/Common/appsvcs/dataStore" } } [Status=Command OK]
Dec  2 15:50:15 bigip2.localhost.localdomain notice mcpd[7403]: 01070417:5: AUDIT - client tmsh, tmsh-pid-21479, user admin - transaction #224744-4 - object 0 - create_if { class_string_item { class_string_item_name "/Common/appsvcs/dataStore" class_string_item_value "as3_async_records0" class_string_item_data "eNrNlk1v3DgMhv+K4UsvVUYf1Fduu1sU6GFPW/SwmSCgJGrGWI/t2p4ARZH/voodoIMEPrSXDuDDWKQoPi8peuq77/u6wxPt69t9HYzkOmjPRDCBQULOPNeKaZ9yBBIhU3F8v6/n5kTTjKehbBOGGw/eGeM4L7ayPp+nJV7sT0NL87pnpOnczs+G7xdOksNqG/puotVYYk94WFPq+ipRbHHEuSkeVe7PXVrixT7RGuDp6en9JQbPEUvCkplskIGQhgXlLNNKWSOsQ4hmEwOcBXUVGJKc5NIBky4nBkoTQy4Ey6Q9GY1JJ7GJoRQocRUY6DMYcop5X/oJojYMExAD0MEBUtJRbmHY8gh7FRiUBFDOmmkXC4ZQnqHi5ZeKRDoYQVFtYWivLMBVYDjhgiBvmNTWMLBaMxRGsBgkBaOUdzFuYGjpuBFXcjesB45ZsURZFIxY2kuWpqIgVc6lUCZtXXHwxnirrwIjQwxkITIJqQxc1JFhLvM3KrI5B+titlsYzgop/VVgeJTeoXVMQIoMJHjmDZhSDa8dZLTW+U0MqYv/T2OAlG8wXhJcTTSO/fjsebevd39i/I+6tPvniCOl3UiPNE7UDTgfH0oHYTng4dBND0O7O9EpFOOO31bTsT+3qTriI1UjfT03ZW81jP1A4/yteheaQzO829f35bQLyT6e2/ZTWjiW/C/FvXCrmqlqukdsm+L7WlDrDCfrEnNaqDJsirRecMc0iCCiVzlg3BRUWP5LXzD+RtAfnndvuyQesTusEY/9NC/LbR+xXd8uG4Y/vdLopV4tTmt+f3z4a9kxxSOd8EvRf3Xa1+rG3vDF1qyaxpZwZLHvcnNY1s9Dwpn+Xs8qIailODeP9JJCN499+wKLYzwWy+cL0coo4ZIzIRmXn6W4VepW2Rsn/b9LVV7VJamUBcbErPe2jB0AhuXqMifK3yMtpC1V26wLf74Tv+u+3tf/AyJ1LN0=" } } [Status=Command OK]
Dec  2 15:50:15 bigip2.localhost.localdomain notice icrd_child[21479]: 01420002:5: AUDIT - pid=21479 user=admin folder=/Common module=(tmos)# status=[Command OK] cmd_data=modify ltm data-group internal /Common/appsvcs/dataStore { records replace-all-with { as3_async_records0 { data eNrNlk1v3DgMhv+K4UsvVUYf1Fduu1sU6GFPW/SwmSCgJGrGWI/t2p4ARZH/voodoIMEPrSXDuDDWKQoPi8peuq77/u6wxPt69t9HYzkOmjPRDCBQULOPNeKaZ9yBBIhU3F8v6/n5kTTjKehbBOGGw/eGeM4L7ayPp+nJV7sT0NL87pnpOnczs+G7xdOksNqG/puotVYYk94WFPq+ipRbHHEuSkeVe7PXVrixT7RGuDp6en9JQbPEUvCkplskIGQhgXlLNNKWSOsQ4hmEwOcBXUVGJKc5NIBky4nBkoTQy4Ey6Q9GY1JJ7GJoRQocRUY6DMYcop5X/oJojYMExAD0MEBUtJRbmHY8gh7FRiUBFDOmmkXC4ZQnqHi5ZeKRDoYQVFtYWivLMBVYDjhgiBvmNTWMLBaMxRGsBgkBaOUdzFuYGjpuBFXcjesB45ZsURZFIxY2kuWpqIgVc6lUCZtXXHwxnirrwIjQwxkITIJqQxc1JFhLvM3KrI5B+titlsYzgop/VVgeJTeoXVMQIoMJHjmDZhSDa8dZLTW+U0MqYv/T2OAlG8wXhJcTTSO/fjsebevd39i/I+6tPvniCOl3UiPNE7UDTgfH0oHYTng4dBND0O7O9EpFOOO31bTsT+3qTriI1UjfT03ZW81jP1A4/yteheaQzO829f35bQLyT6e2/ZTWjiW/C/FvXCrmqlqukdsm+L7WlDrDCfrEnNaqDJsirRecMc0iCCiVzlg3BRUWP5LXzD+RtAfnndvuyQesTusEY/9NC/LbR+xXd8uG4Y/vdLopV4tTmt+f3z4a9kxxSOd8EvRf3Xa1+rG3vDF1qyaxpZwZLHvcnNY1s9Dwpn+Xs8qIailODeP9JJCN499+wKLYzwWy+cL0coo4ZIzIRmXn6W4VepW2Rsn/b9LVV7VJamUBcbErPe2jB0AhuXqMifK3yMtpC1V26wLf74Tv+u+3tf/AyJ1LN0= } } }
Dec  2 15:50:15 bigip2.localhost.localdomain notice mcpd[7403]: 01070417:5: AUDIT - client tmsh, tmsh-pid-21479, user admin - transaction #224762-2 - object 0 - modify { value_list { value_list_name "/Common/appsvcs/dataStore" value_list_config_source 0 } } [Status=Command OK]
Dec  2 15:50:15 bigip2.localhost.localdomain notice mcpd[7403]: 01070417:5: AUDIT - client tmsh, tmsh-pid-21479, user admin - transaction #224762-3 - object 0 - obj_delete { class_string_item { class_string_item_name "/Common/appsvcs/dataStore" } } [Status=Command OK]
Dec  2 15:50:15 bigip2.localhost.localdomain notice mcpd[7403]: 01070417:5: AUDIT - client tmsh, tmsh-pid-21479, user admin - transaction #224762-4 - object 0 - create_if { class_string_item { class_string_item_name "/Common/appsvcs/dataStore" class_string_item_value "as3_async_records0" class_string_item_data "eNrNljuP4zYQgP+Koeaa45qP4Wu7PBAgRaocUmR9WIzIoS1ElhRJXiA4+L9nVlogxm5UJJUBFRJnOJxvXlT19O1QdXimQ/V4qGqnpa1tFKp2tYCMUkRpjbAxlwSk6kKs+PlQzc2ZphnPA29TTroIMTgXpGQZr8+XabGX+vPQ0rzuGWm6tPOr4NuNkpawyoa+m2gVsu0Jj6tLXb/LlFoccW5YY1f6S5cXe6nPtBq4Xq+fbzFkScgOa+GKQwFKO1Gb4IU1xjvlA0JymxgQPJi7wNAUtNQBhA4lCzCWBEqlRCEbyVnMNqtNDGPAqLvAwFjAUTAiRq4nSNYJzEACwNYBkLJNegvD86P8XWBQVkClWGFDYgxlokAj+c0kIls7RclsYdhoPMBdYAQVakXRCW29E+CtFaicEqnWVDtjYkhpA8PqIJ26k97wESQWIzIVxRiJy0tzUVGtTSmcKJe3Whyic9Hbu8AokGrykISGzAMXbRJYeP4mQ76U2odU/BZG8ErreBcYEXUM6INQkJMADVFEB46zEW2Agt6HuImhLev/ZwzQ+gPGm4OriMaxH181nw7V/ntMf1CX97+ecKS8H+mFxom6AefTM1cQ8gHPx256Htr9mc41C/fycTed+kubdyd8od1If14a3rsbxn6gcf5r96lujs3w6VB95dNuQvbTpW1/zgvH4v9tcG/Uds20a7oXbBvWfR9QH5wkH7IIVhkeNhzaqGQQFlStUjSlxrQZUOXl/7rB5IeA/qP59LFK0gm742rx1E/zstz2Cdv167Zg5PVdjN7y1eK0+vfdjz8sO6Z0ojP+xvFflQ6VefAPcpE1a0xTSziK1HelOS7rlyHjTL+sZ7EJainNzQu9udDNY9++weKYTiz5chM0HiVSS6G0kPqLVo/GPBr/EHT8fcnKu7xkk4vClIWP0fPYARDIrSuC4t8jq7TnrG3mRb72xF30K5Ts+A4AoWzgSyBEI3jWaP40KjtM0lP4dwxrpOL77F15DdxbTcfpuH6t/gb10kpg" } } [Status=Command OK]
Dec  2 15:50:15 bigip2.localhost.localdomain notice icrd_child[21479]: 01420002:5: AUDIT - pid=21479 user=admin folder=/Common module=(tmos)# status=[Command OK] cmd_data=modify ltm data-group internal /Common/appsvcs/dataStore { records replace-all-with { as3_async_records0 { data eNrNljuP4zYQgP+Koeaa45qP4Wu7PBAgRaocUmR9WIzIoS1ElhRJXiA4+L9nVlogxm5UJJUBFRJnOJxvXlT19O1QdXimQ/V4qGqnpa1tFKp2tYCMUkRpjbAxlwSk6kKs+PlQzc2ZphnPA29TTroIMTgXpGQZr8+XabGX+vPQ0rzuGWm6tPOr4NuNkpawyoa+m2gVsu0Jj6tLXb/LlFoccW5YY1f6S5cXe6nPtBq4Xq+fbzFkScgOa+GKQwFKO1Gb4IU1xjvlA0JymxgQPJi7wNAUtNQBhA4lCzCWBEqlRCEbyVnMNqtNDGPAqLvAwFjAUTAiRq4nSNYJzEACwNYBkLJNegvD86P8XWBQVkClWGFDYgxlokAj+c0kIls7RclsYdhoPMBdYAQVakXRCW29E+CtFaicEqnWVDtjYkhpA8PqIJ26k97wESQWIzIVxRiJy0tzUVGtTSmcKJe3Whyic9Hbu8AokGrykISGzAMXbRJYeP4mQ76U2odU/BZG8ErreBcYEXUM6INQkJMADVFEB46zEW2Agt6HuImhLev/ZwzQ+gPGm4OriMaxH181nw7V/ntMf1CX97+ecKS8H+mFxom6AefTM1cQ8gHPx256Htr9mc41C/fycTed+kubdyd8od1If14a3rsbxn6gcf5r96lujs3w6VB95dNuQvbTpW1/zgvH4v9tcG/Uds20a7oXbBvWfR9QH5wkH7IIVhkeNhzaqGQQFlStUjSlxrQZUOXl/7rB5IeA/qP59LFK0gm742rx1E/zstz2Cdv167Zg5PVdjN7y1eK0+vfdjz8sO6Z0ojP+xvFflQ6VefAPcpE1a0xTSziK1HelOS7rlyHjTL+sZ7EJainNzQu9udDNY9++weKYTiz5chM0HiVSS6G0kPqLVo/GPBr/EHT8fcnKu7xkk4vClIWP0fPYARDIrSuC4t8jq7TnrG3mRb72xF30K5Ts+A4AoWzgSyBEI3jWaP40KjtM0lP4dwxrpOL77F15DdxbTcfpuH6t/gb10kpg } } }
Dec  2 15:50:16 bigip2.localhost.localdomain notice tmsh[17544]: 01420002:5: AUDIT - pid=17544 user=root folder=/ module=(tmos)# status=[Command OK] cmd_data=cd / ;
Dec  2 15:50:16 bigip2.localhost.localdomain notice tmsh[17544]: 01420002:5: AUDIT - pid=17544 user=root folder=/ module=(tmos)# status=[Command OK] cmd_data=list cm device recursive
Dec  2 15:50:16 bigip2.localhost.localdomain notice icrd_child[21479]: 01420002:5: AUDIT - pid=21479 user=admin folder=/Common module=(tmos)# status=[01020037:3: The requested folder (/Common/appsvcs) already exists.] cmd_data=create sys folder appsvcs
Dec  2 15:50:16 bigip2.localhost.localdomain notice icrd_child[21479]: 01420002:5: AUDIT - pid=21479 user=admin folder=/Common module=(tmos)# status=[01020066:3: The requested value list (/Common/appsvcs/dataStore) already exists in partition Common.] cmd_data=create ltm data-group internal /Common/appsvcs/dataStore { type string }
Dec  2 15:50:16 bigip2.localhost.localdomain notice mcpd[7403]: 01070417:5: AUDIT - client tmsh, tmsh-pid-21479, user admin - transaction #224912-2 - object 0 - modify { value_list { value_list_name "/Common/appsvcs/dataStore" value_list_config_source 0 } } [Status=Command OK]
Dec  2 15:50:16 bigip2.localhost.localdomain notice mcpd[7403]: 01070417:5: AUDIT - client tmsh, tmsh-pid-21479, user admin - transaction #224912-3 - object 0 - obj_delete { class_string_item { class_string_item_name "/Common/appsvcs/dataStore" } } [Status=Command OK]
Dec  2 15:50:16 bigip2.localhost.localdomain notice mcpd[7403]: 01070417:5: AUDIT - client tmsh, tmsh-pid-21479, user admin - transaction #224912-4 - object 0 - create_if { class_string_item { class_string_item_name "/Common/appsvcs/dataStore" class_string_item_value "as3_async_records0" class_string_item_data "eNrNljtv5EYMgP+KoOaao3feD3d5IECKVDmkiPdgUDOcXSFaSZG0BoKD/3volYFb2FGRa24BFdKQw+FHckjVD1/2dY8n2tf3+7pxStjGRpCNa8BkFBCF1WBjLsmQbAqx4sd9vbQnmhc8jbxNOuGiicG5IATLeH05zxd7aTiNHS3rnonmc7e8CL5cKSlhVtk49DOtQrY942F1qR+qTKnDCZeWNaoynPt8sZeGTKuB5+fnj9cYoiRkhxW44hCMVA4aHTxYrb2TPqBJbhPDBG/0TWAoCkqoYECFksFoS4BCSihkIzmL2Wa5iaG10fImMDAW4yhoiJHrySTrALMhMMY2wSBlm9QWhudH+pvAoCwNlWLBhsQYUkdALfhNJyLbOElJb2HYqL0xN4ERZGgkRQfKegfGWwsonYTUKGqc1jGktIFhVRBO3sjd8NEILBoyFckYictLcVFRo3QpnCiXt664ic5Fb28Co5jUkDcJlMnccNEmwML9N2nypTQ+pOK3MIKXSsWbwIioYkAfQJqcwCgTITrjOBvRBlPQ+xA3MZRl/f+NYZR6h/Hq4CqiaRqmF82Hfb37EdNf1Ofd70ecKO8meqJppn7E5fjIFYR8wOOhnx/HbneiU8PCnbiv5uNw7nJ1xCeqJvr73PLeapyGkabln+pD0x7a8cO+/synXYXsl3PX/ZovHBf/r4N7pVa1c9X2T9i1rPs2oD44QT5kCFZqbjYc2ihFAGtkI1PUpcG0GVDpxTdNMPEuoF81H95XSTpif1gtHod5uSx3Q8Ju/bouGPH8Jkav+epwXv374eefLjvmdKQT/sHxX5X2tb7zd+Iia9eYpo5wgjT0pT1c1s9jxoV+W89iE9RRWtonenWhX6ahe4XFKR1Z8ukqaNxKhBIgFQj1Scl7re+1vwsq/nnJypu8ZJ2LxJTBx+i57RgDyFcXguTfIyuV56xt5kW83ImbuK+mZMczwIC0gYdAiBq41yj+1DI7TMJT+G8Mq4XkefbdhsDn+l8tbmtV" } } [Status=Command OK]
Dec  2 15:50:16 bigip2.localhost.localdomain notice icrd_child[21479]: 01420002:5: AUDIT - pid=21479 user=admin folder=/Common module=(tmos)# status=[Command OK] cmd_data=modify ltm data-group internal /Common/appsvcs/dataStore { records replace-all-with { as3_async_records0 { data eNrNljtv5EYMgP+KoOaao3feD3d5IECKVDmkiPdgUDOcXSFaSZG0BoKD/3volYFb2FGRa24BFdKQw+FHckjVD1/2dY8n2tf3+7pxStjGRpCNa8BkFBCF1WBjLsmQbAqx4sd9vbQnmhc8jbxNOuGiicG5IATLeH05zxd7aTiNHS3rnonmc7e8CL5cKSlhVtk49DOtQrY942F1qR+qTKnDCZeWNaoynPt8sZeGTKuB5+fnj9cYoiRkhxW44hCMVA4aHTxYrb2TPqBJbhPDBG/0TWAoCkqoYECFksFoS4BCSihkIzmL2Wa5iaG10fImMDAW4yhoiJHrySTrALMhMMY2wSBlm9QWhudH+pvAoCwNlWLBhsQYUkdALfhNJyLbOElJb2HYqL0xN4ERZGgkRQfKegfGWwsonYTUKGqc1jGktIFhVRBO3sjd8NEILBoyFckYictLcVFRo3QpnCiXt664ic5Fb28Co5jUkDcJlMnccNEmwML9N2nypTQ+pOK3MIKXSsWbwIioYkAfQJqcwCgTITrjOBvRBlPQ+xA3MZRl/f+NYZR6h/Hq4CqiaRqmF82Hfb37EdNf1Ofd70ecKO8meqJppn7E5fjIFYR8wOOhnx/HbneiU8PCnbiv5uNw7nJ1xCeqJvr73PLeapyGkabln+pD0x7a8cO+/synXYXsl3PX/ZovHBf/r4N7pVa1c9X2T9i1rPs2oD44QT5kCFZqbjYc2ihFAGtkI1PUpcG0GVDpxTdNMPEuoF81H95XSTpif1gtHod5uSx3Q8Ju/bouGPH8Jkav+epwXv374eefLjvmdKQT/sHxX5X2tb7zd+Iia9eYpo5wgjT0pT1c1s9jxoV+W89iE9RRWtonenWhX6ahe4XFKR1Z8ukqaNxKhBIgFQj1Scl7re+1vwsq/nnJypu8ZJ2LxJTBx+i57RgDyFcXguTfIyuV56xt5kW83ImbuK+mZMczwIC0gYdAiBq41yj+1DI7TMJT+G8Mq4XkefbdhsDn+l8tbmtV } } }
```

### Expected Behavior
Performing queries of AS3 state should not cause BIG-IP to modify its configuration and cause an out-of-sync state in HA configuration.

### Actual Behavior
BIG-IP goes into "Changes Pending" in HA environment when /info or /declare are queried with a GET method.

**Comment by mdditt2000** [comment]
@euphario looking into this issue.

**Comment by dstokesf5** [comment]
This is caused by AS3 tracking a request history in data groups. The changes pending result is currently expected behavior since the AS3 state is out of sync between devices in the HA pair.

**Comment by mdditt2000** [comment]
PM Jire filed for internal tracking AUTOTOOL-2087

**Comment by walterk82** [comment]
Could you provide docs on how the datagroups are used for internal operation?  i.e. is it safe to delete these and have AS3 re-create them?

For TS, config is stored in the internal local REST storage (which does not seem to be persisted across upgrades).

**Comment by dstokesf5** [comment]
This was resolved in the 3.26.0 release.
